src/EbmlDate.cpp

```diff
@@ -51,6 +51,8 @@ filepos_t EbmlDate::ReadData(IOCallback & input, ScopeMode ReadFully)
     return GetSize();
 
   assert(GetSize() == 8);
+  if (GetSize() != 8)
+    return GetSize();
```

Comment:
This is… different than how other classes handle their cases. Shouldn't there be a "seek to end of element" in the error case?

Reply:
Yes